### PR TITLE
Unify python version choice in bootstrap.sh

### PR DESCRIPTION
**Pull request by cr-marcstevens**
I recently went through python version hell, and I hope this script solves it.
The new `bootstrap.sh` script unifies `bootstrap.sh` and `bootstrap3.sh` as follows:
- `./bootstrap.sh` to use the system's `python`
- `PYTHON=python2 ./bootstrap.sh` to use `python2`
- `PYTHON=python3 ./bootstrap.sh` to use `python3`

It unifies its python version use through the PYTHON environment variable (which defaults to 'python'):
- it uses `$PYTHON` instead of `python`
- it uses `$PYTHON -m pip` instead of `pip`
- it uses `$PYTHON -m virtualenv` instead of `virtualenv`
- it ensures the user's shell uses the same configuration using the `activate` script:
  - it exports the value of environment variable `PYTHON`
  - it aliases `python="$PYTHON"` and `pip="$PYTHON -m pip"`

Other python version issues:
- apparently python2 does not support `-j #` for `python setup.py build_ext`, which I didn't notice because it actually used python3
  so there is an automatic fallback to not using `-j #` with setup.py

I would like your feedback and please test it on some machines with different configurations.

### CODE REVIEW

**Comment by joerowell** [comment]
I will deploy this later across a few different machines and let you know
if I come across any issues - I had some issues with the old build scripts,
so hopefully this will no longer be an issue.

**Comment by lducas** [comment]
Managed compilation and running both with python3 and python2 on:
- Ubuntu 20.04 LTS, Linux 5.4.0-29-generic x86_64

I had to manually install python-tk package. No other issue encountered.
Suggestions:
- update README.rst
- tweak down optimizations for enumeration in fplll compilation, since it is essentially not used from g6k.

**Comment by cr-marcstevens** [comment]
@lducas : both done.
BTW you also mentioned an issue of `python -m virtualenv` not working and a `pip install virtualenv` helped. This is also what I remember having done on `halfpeta` when first starting on that machine. I've added a detection of virtualenv failing and giving a suggestion there in `bootstrap.sh`.

**Comment by joerowell** [comment]
Works for me (with both Python 2 and 3) on the following:
* Ubuntu 18.04 LTS, 5.3.0-51-generic.
* Debian 8.3.0-6, 4.19.0-8-amd64

However, one thing I did notice was that this change isn't currently in rebuild.sh. It would be good to have it there too.

**Comment by cr-marcstevens** [comment]
I was assuming rebuild.sh uses the python alias, or is that not working correctly?
Note that it is not intended to use a different python version with rebuild.sh than that used during setup (specifically it should be the same python version as of fpylll).

**Comment by joerowell** [comment]
A very good point - it appears to use the right python alias, so nothing else is needed (at least I don't see anything!)

**Comment by lducas** [comment]
I guess @malb 's approval is desirable, as the least unknowleageable of us on python's insights.

File: bootstrap.sh [hunk 1]
```diff
@@ -46,7 +74,7 @@ source ./activate
 git clone https://github.com/fplll/fplll g6k-fplll
 cd g6k-fplll || exit
 ./autogen.sh
-./configure --prefix="$VIRTUAL_ENV" $CONFIGURE_FLAGS
+./configure --prefix="$VIRTUAL_ENV" --with-max-parallel-enum-dim=0 $CONFIGURE_FLAGS
```
Comment (malb) [inline bootstrap.sh:77]: I'd not do this, we don't use parallel enumeration much (yet) but this might change and would bias the experience of g6k users.

**Comment by malb** [comment]
Looks good to me otherwise.

**Comment by cr-marcstevens** [comment]
Ok reverted it. So once travis passes, I'll merge it then.

**Comment by malb** [comment]
+1

typed with thumbs, please excuse my brevity

File: bootstrap.sh (hunk 1 above)
Comment (lducas) [reply bootstrap.sh]: Good point !

**Pull request merged**